### PR TITLE
Refactor FlashLight to use an Interface

### PR DESCRIPTION
**Pull request by sps014**
#### Additions made 
```cs
public interface IFlashLight
{
	Task TurnOnAsync();
	Task TurnOffAsync();
}
```

Part of #4497

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/maui?pullRequest=4525) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/maui?pullRequest=4525) before we can accept your contribution.<br/><br/>:x: sps014 [sign now](https://cla.dotnetfoundation.org/dotnet/maui?pullRequest=4525)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/maui?pullRequest=4525) it.</sub>

**Comment by sps014** [comment]
@Redth @mattleibow can you guys please look into it.

File: src/Essentials/src/Flashlight/Flashlight.shared.cs [hunk 1]
```diff
@@ -1,16 +1,34 @@
+#nullable enable
+using System.ComponentModel;
 using System.Threading.Tasks;
+using Microsoft.Maui.Essentials.Implementations;
 
 namespace Microsoft.Maui.Essentials
 {
+	public interface IFlashLight
```
Comment (mattleibow) [inline src/Essentials/src/Flashlight/Flashlight.shared.cs:8]: Can we change the casing to be `IFlashlight`?

**Review by mattleibow — COMMENTED** [review]
Thanks for getting this in here! Looks all good except for the casing of the interface.

**Comment by sps014** [comment]
Thanks for info @mattleibow

**Review by mattleibow — APPROVED** [review]
Excellent stuff!

**Pull request merged**